test/controllers/commitments_controller_test.rb

```diff
@@ -21,10 +21,23 @@ class CommitmentsControllerTest < ActionDispatch::IntegrationTest
 
   test "should GET new" do
     sign_in users(:user_1)
-    get new_commitment_url
+    get new_commitment_url(criterium_id: criteria(:valid_criterium_without_commitment).id)
     assert_response :success
   end
 
+  test "should not GET new without a criterium_id" do
+    sign_in users(:user_1)
+    get new_commitment_url
+    assert_redirected_to new_criterium_path
+  end
+
+  test "should not GET new if the criterium_id matches a different criterium" do
```

Comment:
Would this be clearer as something like "should not GET new if the criterium_id is already associated with a commitment"?

Reply:
changed